tidy3d/material_library/material_library.py

```diff
@@ -80,6 +82,80 @@ def medium(self):
         return self.variants[self.default].medium
 
 
+class Variant2D(Tidy3dBaseModel):
+    """Reference, data_source, and material model for a variant of a 2D material."""
+
+    medium: Medium2D = pd.Field(
+        ...,
+        title="Material dispersion model",
+        description="A dispersive medium described by the pole-residue pair model.",
+    )
+
+    medium_parametric: Callable = pd.Field(
+        None,
+        title="Parametric material model",
+        description="Parameteric 2d material dispersion model.",
+    )
+
+    reference: List[ReferenceData] = pd.Field(
+        None,
+        title="Reference information",
+        description="A list of reference related to this variant model.",
+    )
+
+    data_url: str = pd.Field(
+        None,
+        title="Dispersion data URL",
+        description="The URL to access the dispersion data upon which the material "
+        "model is fitted.",
+    )
+
+
+class Material2D(Tidy3dBaseModel):
```

Comment:
might want @weiliangjin2021  to take a look at this, but it seems to me we can just inherit this from the regular [variant type](https://github.com/flexcompute/tidy3d/blob/849ab3e4930dd38b1ae66cf48939daa367487461/tidy3d/material_library/material_library.py#L51) and overwrite the 'variants' field only? to avoid redundancy

Reply:
Looks to me that `Variant2D` and `Material2D` can inherit from `VariantItem` and `MaterialItem`, respectively. And name them as `Variant2DItem` and `Material2DItem` to be consistent.